### DISCUSSION TITLE
The next release (6.2.60) and betas (6.2.59b)

### DISCUSSION THREAD

**Discussion by ikedas** [sympa-community/sympa#1617] [Release works]
Hi @everyone.
  - Due date of the next stable (6.2.60) is ~~proposed to be 21st Dec. 2020.~~ **postponed to early 2021**.
  - Maybe the first beta (6.2.59b.1) will come out around 24th Nov. 2020.

See also [the milestone 6.2.60](https://github.com/sympa-community/sympa/milestone/15).

----
2020-12-18 update

**Comment by racke** [comment]
This is probably also the release going into Debian 11 (bullseye).

**Comment by ikedas** [comment]
Drafted [release announcement for 6.2.59b.1](https://github.com/sympa-community/sympa/releases) (sign-in needed).  Please check it and add anything you like.

**Comment by ikedas** [comment]
Done!

**Comment by xavierba** [comment]
@ldidry is there any plan for mhonarc.org website and domain ? My understanding is it was quite a bumpy ride to get the SCM, I assume it would be the same for mhonarc.org, but I thought I'd ask.

**Comment by ldidry** [comment]
No, no plan. You may ask, but I don’t know if you will succeed :shrug:

**Comment by racke** [comment]
And if that is even worth it :innocent:

**Comment by ikedas** [comment]
We seem to need one more beta, 6.2.59b.2.  I suggest its date around 7th Dec.

**Comment by ikedas** [comment]
Done!

**Comment by xavierba** [comment]
FWIW, sympa 6.2.59 beta 2 is available in sympa-beta COPR for all supported Fedora and RHEL/CentOS releases.
https://copr.fedorainfracloud.org/coprs/xavierb/sympa-beta

**Comment by ikedas** [comment]
### Release of 6.2.60
**Don't get your priorities wrong.** As announced, release of the next **stable** version of Sympa is planned to be at 21st Dec., 7 days after.  As asked by release announcement of the beta, "_Please install it to test and report bugs, translate user interface to your language, or enhance documentation of Sympa, if you want to help the Sympa community deliver a more reliable version of Sympa._"  **This is the last chance!**

  * Check proposed update of release notes: sympa-community/sympa#1056, and add your comments and fixes as necessity.
      - Most of items need not confirmed as they are auto-generated by github_changelog_generator.  The items in **Changes** should be confirmed and elaborated.

(continue to the next comment)

**Comment by ikedas** [comment]
* Check release announcement: https://github.com/sympa-community/sympa/releases (sign-in needed)

* See description in Upgrading Notes: (Same as "Changes" section in NEWS.md)

**Comment by dverdin** [comment]
What does this sentence mean? "however, listmasters are recommended to review whether wide range of conversion as previous versions is required."

Is it normal the upgrade notes and NEWS.md lead to a 404 ?

**Comment by racke** [comment]
That means that TT tags in the body are no longer evaluated unless you enable them in the configuration. This should be flagged as backward incompatibility.

**Comment by dverdin** [comment]
I know what's in the code change, thank you. It is the release note sentence that is unclear.
Reformulation proposal:

Listmasters are encouraged to review the lists in which personalization is enabled and estimate whether body personalization is really needed before reverting to the previous behaviour.

**Comment by racke** [comment]
My two cents:

_Listmasters are advised to review lists with personalization (merge_feature) enabled and determine whether template variables are used in the body of the emails._ 

That's incomplete, but I strongly suggests that we bring up the documentation up to speed with the new configuration option **before** the release.

**Comment by dverdin** [comment]
Good to me, too.
+1 for documentation.

**Comment by dverdin** [comment]
On second thought.
Though it is an explicit warning, that is not the meaning of Soji's sentence. He wrote just before this sentence that it was possible to revert to the previous behaviour and (I suppose) wanted to question whether body personalization was needed or not before reverting.

So I suggest a slight change:

_however_ listmasters are advised to review lists with personalization (merge_feature) enabled and determine whether template variables _should be_ used in the body of the emails _before reverting_.

The last "before reverting" might be loud, though.

**Comment by ikedas** [comment]
N.B.: I think:
1. Description in [upgrading notes](https://github.com/sympa-community/sympa-community.github.io/blob/master/manual/upgrade/notes.md) would be better to be the most detailed one.
2. Description in [release announcement](https://github.com/sympa-community/sympa/releases) may be a summary text linked to the description above.
3. Description in “Changes” section in release notes may be the same content as 2.

On 1., you (folks discussing here) can edit [the page](https://github.com/sympa-community/sympa-community.github.io/blob/master/manual/upgrade/notes.md) directly, and there are not deadline.

On 2. & 3., please suggest correction (hopefully with patch) on [the corresponding PR](https://github.com/sympa-community/sympa/pull/1056).

**Comment by racke** [comment]
Updated 1. Will add instructions about using that parameter to restore the former glory :grin:

**Comment by ikedas** [comment]
Updated description about setuid in 1.

Updated something in 2 (some links are broken because target pages have not been prepared.). Feel free to add/edit.

**Comment by racke** [comment]
@ikedas I suggest to postpone the release till early next year so we have more time to test the personilization changes and document them properly. My time budget for these tasks are limited until the winter vacation.

**Comment by ikedas** [comment]
Do you mean [this comment](https://github.com/sympa-community/sympa/pull/1038#issuecomment-747291817)?
There is one more issue not yet closed in [Milestone 6.2.60](https://github.com/sympa-community/sympa/milestone/15) which I should confirm.
If you take the former, I may apply to the latter.

**Comment by racke** [comment]
Yes, I can take care of testing the personalization and proper documentation.

**Comment by ikedas** [comment]
I agree to postponement (the date may be decided later).  Thanks for help!

(...So that means, I'll have the time to enjoy [the great conjunction of Jupiter and Saturn](https://www.nasa.gov/feature/the-great-conjunction-of-jupiter-and-saturn) at 21 Dec! :star:)

**Comment by racke** [comment]
> That means that TT tags in the body are no longer evaluated unless you enable them in the configuration. This should be flagged as backward incompatibility.

So that's my suggestion to highlight these kind of changes:

https://github.com/ikedas/sympa/pull/1

**Comment by ikedas** [comment]
### Postponement of release of 6.2.60

Release of 6.2.60 will be postponed for  more reliable debugging and better documentation.

I propose Monday, 4th January 2021 (UTC) as new release date.

**Comment by racke** [comment]
Sounds good to me.

**Comment by racke** [comment]
I plan to run more tests with personalization and writing documentation for it this year, so it will be ready for the 6.2.60 release.

**Comment by ikedas** [comment]
@racke, are changes on translated text possible? If they are, could you please let us know early?  I'll announce postponement to translators' list so that they will be able to update translation.

**Comment by racke** [comment]
Yes I think so. How much time do they need? One week?

**Comment by ikedas** [comment]
I don't know, but one week or more is desirable.

**Comment by racke** [comment]
Ok in that case I refrain from text translations up until the release of 6.2.60, so let us pass it to the translators.

**Comment by ikedas** [comment]
I (hopefully) solved the issue I have deferred (#1036).  From now on I'll check the things reported by the other folks.

**Comment by ikedas** [comment]
### 6.2.60 will be out in 4th Jan

6.2.60 will be out around 4th Jan 2021 08:00 UTC.

**Comment by ikedas** [comment]
Done!

**Comment by racke** [comment]
Thanks a lot. Debian package follows when previous version enters testing (hopefully tomorrow).

**Comment by ikedas** [comment]
Thank you so much helping release of new version!

**Comment by xavierba** [comment]
sympa 6.2.60 as well as mhonarc 2.6.24 are on their way to Fedora/EPEL

**Comment by racke** [comment]
Debian package for Sympa 6.2.60 uploaded. This is the Sympa release which is going to be shipped with Debian 11 (bullseye) unless we need to make an emergency release.
